zulip_bots/zulip_bots/bots/googletranslate/googletranslate.py

```diff
@@ -0,0 +1,104 @@
+# To use this plugin, you need to set up the Google Cloud API key for this bot in
+# googletranslate.conf in this (zulip_bots/bots/googletranslate/) directory.
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import requests
+
+class GoogleTranslateHandler(object):
+    '''
+    This bot will translate any messages sent to it using google translate.
+    Before using it, make sure you set up google api keys, and enable google
+    cloud translate from the google cloud console.
+    '''
+    def usage(self):
+        return '''
+            This plugin allows users translate messages
+            Users should @-mention the bot with the format
+            @-mention "<text_to_translate>" <target-language> <source-language(optional)>
+            '''
+
+    def initialize(self, bot_handler):
+        self.config_info = bot_handler.get_config_info('googletranslate')
+        self.supported_languages = get_supported_languages(self.config_info['key'])
+
+    def handle_message(self, message, bot_handler):
+        bot_response = get_translate_bot_response(message['content'],
+                                                  self.config_info,
+                                                  message['sender_full_name'],
+                                                  self.supported_languages)
+        bot_handler.send_reply(message, bot_response)
+
+api_url = 'https://translation.googleapis.com/language/translate/v2'
+
+help_text = '''
+Google translate bot
+Please format your message like:
+`@-mention "<text_to_translate>" <target-language> <source-language(optional)>`
+Visit [here](https://cloud.google.com/translate/docs/languages) for all languages
+'''
+
+language_not_found_text = 'Language not found. Visit [here](https://cloud.google.com/translate/docs/languages) for all languages'
+
+def get_supported_languages(key):
+    parameters = {'key': key, 'target': 'en'}
+    response = requests.get(api_url + '/languages', params = parameters)
+    if response.status_code == requests.codes.ok:
+        languages = response.json()['data']['languages']
+        return {lang['name'].lower(): lang['language'].lower() for lang in languages}
```

Comment:
This looks better to me, but do you find this change clearer too?

Reply:
Yeah, Just didn't know how to do it before